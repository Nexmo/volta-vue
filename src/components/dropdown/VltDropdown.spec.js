import { shallowMount } from '@vue/test-utils';
import { VltDropdown } from '..';

describe('vlt-dropdown', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltDropdown);
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders correctly with label "Pick one"', () => {
    wrapper.setProps({
      label: 'Pick one',
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('can open dropdown panel with basic options', () => {
    wrapper.setProps({
      options: ['red', 'yellow', 'blue'],
    });
    wrapper.find('button').trigger('click');
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders fine when unbordered is true and trigger is true', () => {
    wrapper.setProps({
      options: ['red', 'yellow', 'blue'],
      unbordered: true,
      trigger: true,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('when "showSelection" is true, button text is "yellow"', () => {
    wrapper.setProps({
      options: ['red', 'yellow', 'blue'],
      showSelection: true,
    });
    wrapper.find('button').trigger('click');
    wrapper.findAll('.Vlt-dropdown__link').at(1).trigger('click');

    expect(wrapper.find('.Vlt-dropdown__btn').text()).toBe('yellow');
    expect(wrapper.vm.selectedOption).toBe('yellow');
  });

  test('when has label and "showSelection" is false, button text is "Pick one"', () => {
    wrapper.setProps({
      options: ['red', 'yellow', 'blue'],
      label: 'Pick one',
    });
    wrapper.find('button').trigger('click');
    wrapper.findAll('.Vlt-dropdown__link').at(1).trigger('click');

    expect(wrapper.find('.Vlt-dropdown__btn').text()).toBe('Pick one');
    expect(wrapper.vm.selectedOption).toBe('yellow');
  });

  test('when has label, "showSelection" is true and "hideLabel" is true,  dropdown button has text "yellow"', () => {
    wrapper.setProps({
      options: ['red', 'yellow', 'blue'],
      label: 'Pick one',
      showSelection: true,
      hideLabel: true,
    });
    wrapper.find('button').trigger('click');
    wrapper.findAll('.Vlt-dropdown__link').at(1).trigger('click');

    expect(wrapper.find('.Vlt-dropdown__btn').text()).toBe('yellow');
    expect(wrapper.vm.selectedOption).toBe('yellow');
  });

  test('when has label, "showSelection" is true and "hideLabel" is false,  dropdown button has text "Pick one: yellow"', () => {
    wrapper.setProps({
      options: ['red', 'yellow', 'blue'],
      label: 'Pick one',
      showSelection: true,
    });
    wrapper.find('button').trigger('click');
    wrapper.findAll('.Vlt-dropdown__link').at(1).trigger('click');

    expect(wrapper.find('.Vlt-dropdown__btn').text().replace(/[\s]{2,}/g, ' ')).toContain('Pick one: yellow');
    expect(wrapper.vm.selectedOption).toBe('yellow');
  });
});
