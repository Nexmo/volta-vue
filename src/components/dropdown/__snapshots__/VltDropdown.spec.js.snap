// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-dropdown can open dropdown panel with basic options 1`] = `
<div
  class="Vlt-dropdown Vlt-dropdown--expanded"
>
  <button
    class="Vlt-dropdown__btn"
  >
    <!---->
     
    <!---->
  </button>
   
  <div
    class="Vlt-dropdown__panel"
  >
    <div
      class="Vlt-dropdown__panel__content"
    >
      <ul>
        <li
          id="undefined-redDropdownLink"
        >
          <a
            class="Vlt-dropdown__link"
          >
            <span
              class="Vlt-dropdown__label"
            >
              red
            </span>
          </a>
        </li>
        <li
          id="undefined-yellowDropdownLink"
        >
          <a
            class="Vlt-dropdown__link"
          >
            <span
              class="Vlt-dropdown__label"
            >
              yellow
            </span>
          </a>
        </li>
        <li
          id="undefined-blueDropdownLink"
        >
          <a
            class="Vlt-dropdown__link"
          >
            <span
              class="Vlt-dropdown__label"
            >
              blue
            </span>
          </a>
        </li>
         
        <!---->
      </ul>
    </div>
  </div>
</div>
`;

exports[`vlt-dropdown renders correctly 1`] = `
<div
  class="Vlt-dropdown"
>
  <button
    class="Vlt-dropdown__btn"
  >
    <!---->
     
    <!---->
  </button>
   
  <div
    class="Vlt-dropdown__panel"
  >
    <div
      class="Vlt-dropdown__panel__content"
    >
      <ul>
         
        <!---->
      </ul>
    </div>
  </div>
</div>
`;

exports[`vlt-dropdown renders correctly with label "Pick one" 1`] = `
<div
  class="Vlt-dropdown"
>
  <button
    class="Vlt-dropdown__btn"
  >
    <span>
      
        Pick one
      <!---->
    </span>
     
    <!---->
  </button>
   
  <div
    class="Vlt-dropdown__panel"
  >
    <div
      class="Vlt-dropdown__panel__content"
    >
      <ul>
         
        <!---->
      </ul>
    </div>
  </div>
</div>
`;

exports[`vlt-dropdown renders fine when unbordered is true and trigger is true 1`] = `
<div
  class="Vlt-dropdown Vlt-dropdown__trigger Vlt-dropdown__trigger--btn"
>
  <button
    class="Vlt-btn Vlt-btn--tertiary Vlt-btn-- Vlt-btn--unbordered"
  >
    <!---->
     
    <!---->
  </button>
   
  <div
    class="Vlt-dropdown__panel"
  >
    <div
      class="Vlt-dropdown__panel__content"
    >
      <ul>
        <li
          id="undefined-redDropdownLink"
        >
          <a
            class="Vlt-dropdown__link"
          >
            <span
              class="Vlt-dropdown__label"
            >
              red
            </span>
          </a>
        </li>
        <li
          id="undefined-yellowDropdownLink"
        >
          <a
            class="Vlt-dropdown__link"
          >
            <span
              class="Vlt-dropdown__label"
            >
              yellow
            </span>
          </a>
        </li>
        <li
          id="undefined-blueDropdownLink"
        >
          <a
            class="Vlt-dropdown__link"
          >
            <span
              class="Vlt-dropdown__label"
            >
              blue
            </span>
          </a>
        </li>
         
        <!---->
      </ul>
    </div>
  </div>
</div>
`;
